perceval/rendering/canvas/canvas.py

```diff
@@ -262,6 +262,31 @@ def add_text(self, points: tuple[float, float],
             self.position = (points[0]+size*len(text)/2, points[1]+size)
         return f_points[0], self._inverse_Y * f_points[1]
 
+    def normalize_text(self, text, size, points, max_size):
+        if max_size is not None and size * len(text) > max_size:
+            new_size = int(max_size / len(text) * 1.5)
+            if new_size < size / 2:
+                size = size / 2
+                # try to split text on spaces instead of reducing size
+                pos_x, pos_y = points
+                remaining = text
+                new_text = ''
+                min_index = int(max_size/size * 0.8)
```

Comment:
The use of a min index here is strange.
For example, if the first space is before the min index and the second is after the max index, the text will be cut in the middle of the intermediate word.
I think it is better to split on spaces and change the line if the next word would make the number of characters above the limit.